flink-tests/src/test/scala/org/apache/flink/api/scala/migration/StatefulJobWBroadcastStateMigrationITCase.scala

```diff
@@ -75,43 +75,65 @@ object StatefulJobWBroadcastStateMigrationITCase {
       SnapshotSpec.withVersions(
         StateBackendLoader.MEMORY_STATE_BACKEND_NAME,
         SnapshotType.SAVEPOINT_CANONICAL,
-        FlinkVersion.rangeOf(FlinkVersion.v1_8, FlinkVersion.v1_13)))
+        if (targetGeneratingVersion != null)
```

Comment:
Defining a local BiFunction would shorten the code even more
```
    var getFlinkVersions =
      new BiFunction[FlinkVersion, FlinkVersion, util.Collection[FlinkVersion]] {
        override def apply(
            minInclVersion: FlinkVersion,
            maxInclVersion: FlinkVersion): util.Set[FlinkVersion] = if (
          targetGeneratingVersion != null
        )
          Collections.singleton(targetGeneratingVersion)
        else
          FlinkVersion.rangeOf(minInclVersion, maxInclVersion)
      }
```
All the changes below could be replaced with a `getFlinkVersions` which reduces the redundant code.